pkg/publisher/webhookPublisher.go

```diff
@@ -80,52 +80,53 @@ func (p *WebhookPublisher) svc() {
 
 func (p *WebhookPublisher) makeHttpRequest(r *publishRequest) {
 	url := p.baseUrl + "/" + strings.TrimPrefix(r.target, "/")
-	p.logger.Info("making HTTP request", zap.String("url", url))
+
+	msg := "making HTTP request"
+	level := zap.InfoLevel
+	fields := []zap.Field{zap.String("url", url), zap.String("type", "publish_request")}
+
+	// log once for this request
+	defer func() {
+		if ce := p.logger.Check(level, msg); ce != nil {
+			ce.Write(fields...)
+		}
+	}()
 
 	var buf bytes.Buffer
 	buf.WriteString(r.body)
 
 	// Create request
 	req, err := http.NewRequest("POST", url, &buf)
 	if err != nil {
-		p.logger.Error("error creating request", zap.Error(err), zap.String("url", url))
+		level = zap.ErrorLevel
+		fields = append(fields, zap.Error(err))
+		return
 	}
 	for k, v := range r.headers {
 		req.Header.Set(k, v)
 	}
 
 	// Make the request
 	resp, err := http.DefaultClient.Do(req)
-
-	// All done if the request succeeded with 200 OK.
-	if err == nil && resp.StatusCode == 200 {
+	var body []byte
+	if err == nil {
```

Comment:
Though I can understand the whole code flow, however, 

1. two `if err == nil` here may confuse people when code reading.
2. people may miss the err of `ioutil.ReadAll` inside block. 

How about we check the `err != nil` first, so that we can move `ioutil.ReadAll` out from the block?

Reply:
Code has been changed as you suggested, hope u review again, thx!